core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/BlockletDataMapFactory.java

```diff
@@ -315,33 +297,64 @@ public void clear(Segment segment) {
   public synchronized void clear() {
     if (segmentMap.size() > 0) {
       for (String segmentId : segmentMap.keySet().toArray(new String[segmentMap.size()])) {
-        clear(new Segment(segmentId, null, null));
+        clear(segmentId);
       }
     }
   }
 
-  @Override
-  public List<CoarseGrainDataMap> getDataMaps(DataMapDistributable distributable)
+  @Override public List<String> getAllTableBlockletUniqueNames() {
```

Comment:
Better return the cache size used by each datamap from here directly instead of getting the index files and looking onto cache.

Reply:
done